transform/mattermost-analytics/models/marts/sales/hightouch/fct_in_product_trial_requests.sql

```diff
@@ -0,0 +1,48 @@
+{% set size_buckets = {
+    'ONE_TO_50': 1,
+    'FIFTY_TO_100': 50,
+    'ONE_HUNDRED_TO_500': 100,
+    'FIVE_HUNDRED_TO_1000': 500,
+    'ONE_THOUSAND_TO_2500': 1000,
+    'TWO_THOUSAND_FIVE_HUNDRED_AND_UP': 2500
+}
+%}
+
+with trial_requests as (
+    select
+        trial_request_id,                                   -- Primary key
+        -- Convention followed in SF
+        coalesce(
+            name,
+            left(email, 40)
+        ) as name,                                          -- Mapped to field name of lead
+        first_name,                                         -- Mapped to field first_name of lead
+        last_name,                                          -- Mapped to field last_name of lead
+        case
+        {% for bucket, size_lower in size_buckets.items() -%}
+            when company_size_bucket = '{{bucket}}' then {{size_lower}}
+        {% endfor -%}
+        end as company_size,                                -- Mapping lower threshold
+        company_name,                                       -- Mapped to field company of lead
+        -- Salesforce lowercases email
+        lower(coalesce(contact_email, email)) as email,     -- Mapped to field email of lead
+        case
+           -- Fix inconsistency in country name so that it matches values expected by SF
+           when country_name = 'United States of America' then 'United States'
+           else country_name
+        end as country_name,                               -- Mapped to field country
+        start_at as trial_start_at,                         -- Mapped to request_a_trial_date__c and TBD
+        num_users                                           -- Mapped to field numberofemployees of lead
+    from
+        {{ ref('stg_cws__trial_requests')}}
+)
+select
```

Comment:
According to Kevin's latest comments - https://mattermost.atlassian.net/browse/MM-51085?focusedCommentId=149587 - looks like we need to create a campaign member as well(?) Also, according to John's description - "a lead entry should be created in SFDC if one does not already exist" - is that done by a different model?

Reply:
There's a few things to clarify in today's meeting. I'll add them in the ticket. 

Regarding new vs existing items, a single model with a flag on whether a lead with the same email exists should be enough. Hightouch allows for updates based on models or queries. My idea was to have a single model and use the flag on hightouch to decide on whether an update or insert should happen.